### PR TITLE
OTIO 0.17.0 Release Prep

### PR DESCRIPTION
**Pull request by markreidvfx**
Preparing the AAF adapter repo for release of OpenTimelineIO 0.17.0 that ships without adapters.

I tried my best to keep this somewhat matching the style of the ALE adapter. This repo is different in that it uses a wheel for all the tests. This is so we are testing the same thing that will users will install.

I also took the opportunity to convert the `setup.py` to `pyproject.toml`

I tried to update the `codecov-action` to `v4` but they require some sort of token now that I'm not sure how to generate yet.

### CODE REVIEW

**Comment by meshula** [comment]
Note to @markreidvfx and reviewers: when landing this one please Merge, don't use Squash and Merge; it would be helpful to keep [0849fc2](https://github.com/OpenTimelineIO/otio-aaf-adapter/pull/33/commits/0849fc23c8b0294744732f90b3a8dd9cb0876e9e) as a separate commit in history.

**Review by meshula — APPROVED** [review]
lgtm, approving pending updates per @apetrynet

**Comment by markreidvfx** [comment]
@apetrynet it should be all good now, sorry for the delay. I removed the `.dev1` like you suggested. I also added they dynamic badges, I didn't know that was a thing now :). I'll remove using `@extract_adapters` branch once it has been merged into the main OpenTimelineIO repo. I also rebased it on top of the current main.

**Review by apetrynet — APPROVED** [review]

**Pull request merged**
